Description: Remove backup files
 Dia normally saves a copy of an old saved file if it is overwritten via the
 "File Save" function. This clutters the directory. Since the common behaviour
 of programs is to not save backups in a separate location (also for security
 reasons), this patch removes this feature and therefore removes backup files
 when dia is sure to have the newly written file in the correct location.
Author: Roland Stigge <stigge@antcom.de>
Bug-Debian: http://bugs.debian.org/173650

--- dia-0.97.2.orig/app/load_save.c
+++ dia-0.97.2/app/load_save.c
@@ -1078,6 +1078,8 @@ diagram_data_save(DiagramData *data, con
 		  dia_message_filename(filename), 
 		  dia_message_filename(filename), strerror(errno));
   }
+  else /* remove backup file if new file is in place */
+    g_unlink(bakname);
 CLEANUP:
   if (filename != user_filename)
     g_free(filename);
